Wrappers/Python/cil/processors/PaganinProcessor.py

```diff
@@ -0,0 +1,405 @@
+            if parameter not in parameters:
+                raise ValueError('Parameter {} not recognised, expected one of {}.'.format(parameter, parameters))
+            elif (override_geometry[parameter] is None) | (override_geometry[parameter] == 0):
+                raise ValueError("Parameter {} cannot be {}, please update data.geometry.{} or over-ride with processor.get_output(override_geometry= {{ '{}' : value }} )"\
+                    .format(parameter, str(getattr(self, parameter)), geometry_parameters[i], parameter))
+            else:
+                self.__setattr__(parameter, override_geometry[parameter])
+
+
+        # get and check parameters from geometry if they are not in the over-ride geometry dictionary
+        for i, parameter in enumerate(parameters):
+            if parameter not in override_geometry:
+                if type(geometry_parameters[i])==tuple:
+                    param1 = getattr(geometry, geometry_parameters[i][0])
+                    param2 = getattr(geometry, geometry_parameters[i][1])
+                    if (param1 - param2) / (param1 + param2) >= 1e-5:
+                        raise ValueError("Parameter {} is not homogeneous up to 1e-5: got {} and {}, please update geometry using data.geometry.{} and data.geometry.{}\
+                                            or over-ride with processor.get_output(override_geometry={{ '{}' : value }} )"\
+                                            .format(parameter, str(param1), str(param2), geometry_parameters[i][0], geometry_parameters[i][1], parameter))
+                else:
+                    param1 = getattr(geometry, geometry_parameters[i])
+                
+                if (param1 is None) | (param1 == 0):
+                    raise ValueError("Parameter {} cannot be {}, please update data.geometry.{} or over-ride with processor.get_output(override_geometry={{ '{}' : value }} )"\
+                    .format(parameter, str(param1) ,str(geometry_parameters[i]), parameter))
+                else:
+                    if convert_units[i]:
+                        if unit_multiplier is None:
+                            raise ValueError("Geometry units {} not recognised, expected one of {}, please update data.geometry.config.units".format(str(geometry.config.units),str(self.UNIT_LIST)))
+                        else:
+                            self.__setattr__(parameter, param1*unit_multiplier)
+                    else:
+                        self.__setattr__(parameter, param1)
+        
+    def _create_filter(self, Nx, Ny, override_filter=None):
+        '''
+        Function to create the Paganin filter, either using the paganin [1] or generalised paganin [2] method
+        The filter is created on a mesh in Fourier space kx, ky
+        [1] https://doi.org/10.1046/j.1365-2818.2002.01010.x
+        [2] https://iopscience.iop.org/article/10.1088/2040-8986/abbab9 
+        '''
+        if override_filter is None:
```

Comment:
A quick comment line on each part of creating the filter would improve readability.

Reply:
Comments added.